sinks/api/endpoint_test.go

```diff
@@ -0,0 +1,801 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+// Adapted for Orb project, modifications licensed under MPL v. 2.0:
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/gofrs/uuid"
+	mfsdk "github.com/mainflux/mainflux/pkg/sdk/go"
+	"github.com/mainflux/mainflux/things"
+	thmocks "github.com/mainflux/mainflux/things/mocks"
+	"github.com/ns1labs/orb/pkg/types"
+	"github.com/ns1labs/orb/sinks"
+	skmocks "github.com/ns1labs/orb/sinks/mocks"
+	"github.com/opentracing/opentracing-go/mocktracer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+const (
+	contentType  = "application/json"
+	token        = "token"
+	invalidToken = "invalid"
+	email        = "user@example.com"
+	validJson    = "{\n    \"name\": \"my-prom-sink\",\n    \"backend\": \"prometheus\",\n    \"config\": {\n        \"remote_host\": \"my.prometheus-host.com\",\n        \"username\": \"dbuser\"\n    },\n    \"description\": \"An example prometheus sink\",\n    \"tags\": {\n        \"cloud\": \"aws\"\n    },\n    \"validate_only\": false\n}"
+	invalidJson  = "{"
+)
+
+var (
+	nameID, _ = types.NewIdentifier("my-sink")
+	sink      = sinks.Sink{
+		Name:        nameID,
+		Description: "An example prometheus sink",
+		Backend:     "prometheus",
+		Config:      map[string]interface{}{"remote_host": "data", "username": "dbuser"},
+		Tags:        map[string]string{"cloud": "aws"},
+	}
+	invalidName  = strings.Repeat("m", maxNameSize+1)
+	notFoundRes  = toJSON(errorRes{things.ErrNotFound.Error()})
+	unauthRes    = toJSON(errorRes{things.ErrUnauthorizedAccess.Error()})
+	notSupported = toJSON(errorRes{sinks.ErrUnsupportedContentTypeSink.Error()})
+	wrongID, _   = uuid.NewV4()
+)
+
+type testRequest struct {
+	client      *http.Client
+	method      string
+	url         string
+	contentType string
+	token       string
+	body        io.Reader
+}
+
+func (tr testRequest) make() (*http.Response, error) {
+	req, err := http.NewRequest(tr.method, tr.url, tr.body)
+	if err != nil {
+		return nil, err
+	}
+	if tr.token != "" {
+		req.Header.Set("Authorization", tr.token)
+	}
+	if tr.contentType != "" {
+		req.Header.Set("Content-Type", tr.contentType)
+	}
+	return tr.client.Do(req)
+}
+
+func newService(tokens map[string]string) sinks.Service {
+	auth := thmocks.NewAuthService(tokens)
```

Comment:
this brings in a dependency on mainflux things, we should probably fork this if we need it